workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -198,6 +201,20 @@ def _get_and_execute_task_graph(self, token_info, deployment_id,
 
         self._semaphore.release()
 
+    def _possibly_update_encryption_key(self):
+        with open(SECURITY_FILE_LOCATION) as security_conf_file:
+            rest_security_conf = json.load(security_conf_file)
+        enc_key = base64.urlsafe_b64decode(str(
+            rest_security_conf['encryption_key'],
+        ))
+        if len(enc_key) == 32:
```

Comment:
Older managers only support 32 byte keys. If the key length isn't 32 bytes, we'd crash trying to update the key. I'm not sure if that's what we want to do or not, since it would indicate we were restoring from a manager that had already got broken secrets.